Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -345,19 +350,30 @@ void RunGpuLoop()
 				// If we don't, s_swapRequested or s_efbAccessRequested won't be set to false
 				// leading the CPU thread to wait in Video_BeginField or Video_AccessEFB thus slowing things down.
 				AsyncRequests::GetInstance()->PullEvents();
-				CommandProcessor::isPossibleWaitingSetDrawDone = false;
 			}
-
 			fifo.isGpuReadingData = false;
+			s_gpu_done_event.Set();
```

Comment:
Reminder for mysef: This has a race condition when new data is pushed right now before this event is set, the next call to FlushGpu won't block.